File: Ruby/kex_2.rb
require_relative 'kex_2_battle'
require_relative 'kex_2_enemy'

class Kex_2
  def execute
    puts '冒険が今始まる！'
    player = PlayerEx.new
    continue_game = 1
    kills = 0

    while continue_game != 0
      puts "現HP：#{player.hp}"
      print '奥に進みますか？（１：奥に進む　０．帰る）＞'
      continue_game = gets.chomp.to_i
      if continue_game == 1
        type_no = rand(0..2)
        enemy = create_enemy(type_no)
        result = attack_enemy(player, enemy)
        kills += 1
        
        unless result
          break
        end
      end
    end

    puts "リ〇ミト！\n"
    puts "戦闘回数：#{kills}回　残りHP：#{player.hp}"
  end
end

File: Cpp/Kex_2.cpp
#include "Kex_2.h"
#include <iostream>
#include "Kex_2_Battle.h"
#include "Kex_2_Enemy.h"

using std::cout;

void Kex_2::Execute()
{
	auto engine = Kex_2::GetEngine<>();
	std::uniform_int_distribution<> generator(0, 2);
	cout << "冒険が今始まる！\n";
	PlayerEx player;
	player.Hp = PLAYER_INITIAL_HP;
	player.Defense = PLAYER_DEFENSE;
	int continue_game = 1;
	int kills = 0;

	do
	{
		cout << "\n現HP：" << player.Hp << "\n";
		cout << "奥に進みますか？（１：奥に進む　０．帰る）＞";
		std::cin >> continue_game;
		if (continue_game == 1)
		{
			int lv = generator(engine);
			Enemy enemy = Kex_2_Enemy::CreateEnemy(static_cast<EnemyType>(lv));
			bool result = Kex_2_Battle::AttackEnemy(player, enemy);
			++kills;
			if (!result)
			{
				break;
			}
		}

	} while (continue_game);

	cout << "リ〇ミト！\n\n";
	cout << "戦闘回数：" << kills << "回　残りHP：" << player.Hp << "\n";
}
